feature/experimental/platform/tests/optics_thresholds_test/optics_thresholds_test.go

```diff
@@ -0,0 +1,239 @@
+	}, {
+		desc:         "Check laser-temperature-low-warn-threshold",
+		property:     "laser-temperature-low-warn-threshold",
+		minThreshold: minOpticsTemperature,
+		maxThreshold: maxOpticsTemperatureLowThreshold,
+	}, {
+		desc:         "Check laser-temperature-low-alarm-threshold",
+		property:     "laser-temperature-low-alarm-threshold",
+		minThreshold: minOpticsTemperature,
+		maxThreshold: maxOpticsTemperatureLowThreshold,
+	}, {
+		desc:         "Check laser-bias-current-high-warn-threshold",
+		property:     "laser-bias-current-high-warn-threshold",
+		minThreshold: minOpticsBiasCurrentHighThreshold,
+		maxThreshold: maxOpticsBiasCurrent,
+	}, {
+		desc:         "Check laser-bias-current-high-alarm-threshold",
+		property:     "laser-bias-current-high-alarm-threshold",
+		minThreshold: minOpticsBiasCurrentHighThreshold,
+		maxThreshold: maxOpticsBiasCurrent,
+	}, {
+		desc:         "Check laser-bias-current-low-warn-threshold",
+		property:     "laser-bias-current-low-warn-threshold",
+		minThreshold: minOpticsBiasCurrent,
+		maxThreshold: maxOpticsBiasCurrentLowThreshold,
+	}, {
+		desc:         "Check laser-bias-current-low-alarm-threshold",
+		property:     "laser-bias-current-low-alarm-threshold",
+		minThreshold: minOpticsBiasCurrent,
+		maxThreshold: maxOpticsBiasCurrentLowThreshold,
+	}}
+
+	for _, tc := range cases {
+		t.Log(tc.desc)
+		t.Run(tc.desc, func(t *testing.T) {
+			for _, transceiver := range transceivers {
+				t.Logf("Validate transceiver: %s", transceiver)
+				component := dut.Telemetry().Component(transceiver)
+				if !component.MfgName().Lookup(t).IsPresent() {
+					t.Logf("component.MfgName().Lookup(t).IsPresent() for %q is false. skip it", transceiver)
+					continue
+				}
+				mfgName := component.MfgName().Get(t)
+				t.Logf("Transceiver %s MfgName: %s", transceiver, mfgName)
+
+				threshold := fetchOpticsThreshold(t, dut, transceiver, tc.property)
+				if threshold > tc.maxThreshold || threshold < tc.minThreshold {
+					t.Errorf("Get threshold for %q): got %.2f, want within [%f, %f] ", transceiver, threshold, tc.minThreshold, tc.maxThreshold)
+				}
+			}
+		})
+	}
+}
+
+func fetchOpticsThreshold(t *testing.T, dut *ondatra.DUTDevice, opticsName string, property string) float64 {
+	t.Helper()
+	// TODO: Need to update the lookup code after optics threshold model is defined.
+	t.Skipf("Optics threshold model needs to be defined, skip it for now.")
+
+	val := dut.Telemetry().Component(opticsName).Property(property).Get(t).GetValue()
+	switch v := val.(type) {
+	case telemetry.UnionUint64:
+		return float64(v)
+	case telemetry.UnionInt64:
+		return float64(v)
+	case telemetry.UnionFloat64:
+		return float64(v)
+	default:
+		t.Fatalf("Error extracting optics threshold, could not type assert union. union: %v", val)
+		return 0
+	}
+}
+
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
```

Comment:
(non-blocking) Is this a duplicate from somewhere else? If so, should there be a common internal library for doing this?

Reply:
Yes, we are working on helper functions.